bindings/python/setup.py

```diff
@@ -136,36 +140,86 @@ def write_b2_python_config(config):
+    CONFIG_MODE_B2 = "b2"
+    CONFIG_MODES = (CONFIG_MODE_DISTUTILS, CONFIG_MODE_B2)
+
     user_options = BuildExtBase.user_options + [
+        (
+            "config-mode=",
+            None,
+            "'b2' or 'distutils' (default). "
+            "In b2 mode, setup.py will just invoke b2 using --b2-args. "
+            "It will not attempt to auto-configure b2 or override any "
+            "args. "
+            "In distutils mode, setup.py will attempt to configure "
+            "and invoke b2 to match the expectations and behavior of "
+            "distutils (libtorrent will be built against the invoking "
+            "python, etc; note not all behaviors are currently supported). "
+            "The feature set will match the version found on pypi. "
+            "You can selectively override the auto-configuration in "
+            "this mode with --no-autoconf or --b2-args. For example "
+            "--b2-args=python=x.y or --no-autoconf=python will prevent "
+            "python from being auto-configured. "
+            "Note that --b2-args doesn't currently understand implicit features. "
+            "Be sure to include their names, e.g. --b2-args=variant=debug",
+        ),
+        (
+            "b2-args=",
+            None,
+            "The full argument string to pass to b2. This is parsed with shlex, to "
+            "support arguments with spaces. For example: --b2-args 'variant=debug "
+            '"my-feature=a value with spaces"\'',
+        ),
+        (
+            "no-autoconf=",
+            None,
+            "Space-separated list of b2 arguments that should not be "
+            "auto-configured in distutils mode.",
+        ),
         (
             "libtorrent-link=",
             None,
-            "how to link to libtorrent ('static', 'shared' or 'prebuilt')",
+            "(DEPRECATED; use --b2-args=libtorrent-link=...) ",
         ),
         (
             "boost-link=",
             None,
-            "how to link to boost-python ('static' or 'shared')",
+            "(DEPRECATED; use --b2-args=boost-link=...) "
+        ),
+        ("toolset=", None, "(DEPRECATED; use --b2-args=toolset=...) b2 toolset"),
+        (
+            "pic",
+            None,
+            "(DEPRECATED; use --b2-args=libtorrent-python-pic=on) "
+            "whether to compile with -fPIC",
+        ),
+        (
+            "optimization=",
+            None,
+            "(DEPRECATED; use --b2-args=optimization=...) " "b2 optimization mode",
         ),
-        ("toolset=", None, "b2 toolset"),
-        ("pic", None, "whether to compile with -fPIC"),
-        ("optimization=", None, "b2 optimization mode"),
         (
-            "hash",
+            "cxxstd=",
             None,
-            "use a property hash for the build directory, rather than "
-            "property subdirectories",
+            "(DEPRECATED; use --b2-args=cxxstd=...) "
+            "boost cxxstd value (14, 17, 20, etc.)",
         ),
```

Comment:
Do we not want to keep `--hash`? I figured this might get used on windows

Reply:
yeah, I didn't mean to drop this